rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -39,14 +39,22 @@ class NodeBase : public NodeBaseInterface, public std::enable_shared_from_this<N
 public:
   RCLCPP_SMART_PTR_ALIASES_ONLY(NodeBase)
 
+  /// Constructor.
+  /**
+   * If nullptr (default) is given for the default_callback_group, one will
+   * be created by the constructor using the create_callback_group() method,
+   * but virtual dispatch will not occur so overrides of that method will not
+   * be used.
+   */
   RCLCPP_PUBLIC
   NodeBase(
     const std::string & node_name,
     const std::string & namespace_,
     rclcpp::Context::SharedPtr context,
     const rcl_node_options_t & rcl_node_options,
     bool use_intra_process_default,
-    bool enable_topic_statistics_default);
+    bool enable_topic_statistics_default,
+    rclcpp::CallbackGroup::SharedPtr default_callback_group = nullptr);
```

Comment:
is this related to the PR?

Reply:
It seems okay though, just trying to understand if it's an unrelated addition or it's related to the virtual dispatch in constructor/destructor issue.

Reply:
It's related, as I need to pass the default callback group in order to avoid the issues with calling virtual methods in the destructor.